src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -277,8 +289,52 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
                     }
                 }
             }
+        }
+
+        private static void AddNamesFromExistingOverloads(CSharpSyntaxContext context, SemanticModel semanticModel, ArrayBuilder<(string name, SymbolKind kind)> result, CancellationToken cancellationToken)
+        {
+            var namedType = semanticModel.GetEnclosingNamedType(context.Position, cancellationToken);
+            if (namedType is null)
+                return;
+
+            var parameterSyntax = context.LeftToken.GetAncestor(n => n.IsKind(SyntaxKind.Parameter)) as ParameterSyntax;
+            if (parameterSyntax is not { Type: { } parameterType, Parent: { Parent: BaseMethodDeclarationSyntax baseMethod } })
+                return;
```

Comment:
can you check that the left token is hte last token of the parameter type.  i'm actually worried about you offering these names *inside* the parameter type.  Can you add tests that shows this doesn't happen (for example, in the type arg of a generic type).

Reply:
@CyrusNajmabadi Likely already handled in `GetDeclarationInfoAsync`. I'm not sure about the exact test case you're referring to. Something like this?

```csharp
void M<T>($$) { }
```

Reply:
No. more like:  `void M(List<$$>`  the left token will be inside a parameter here.